bugsnag-android-core/src/test/java/com/bugsnag/android/MetadataConcurrentModificationTest.kt

```diff
@@ -18,4 +19,26 @@ internal class MetadataConcurrentModificationTest {
             }
         }
     }
+
+    @Test()
+    fun testConcurrentModificationRedactedKeys() {
+        val keys = mutableSetOf("alpha", "omega")
+        val orig = Metadata()
+        val executor = Executors.newSingleThreadExecutor()
+
+        repeat(100) {
+            orig.redactedKeys = keys
+            executor.execute {
+                keys.add("$it")
+            }
+        }
+    }
```

Comment:
Some comments would be useful here for future context as it's a non-obvious test.

Reply:
Added some KDoc